launch_frontend/launch_frontend/parser.py

```diff
@@ -0,0 +1,213 @@
+
+
+def get_attr_or_default(entity, attr, default_value):
+    """Return the attr value of entity or default_value."""
+    try:
+        return entity.__getattr__(attr)
+    except AttributeError:
+        return default_value
+
+
+def get_attr_or_none(entity, attr):
+    """Return the attr value of entity or None."""
+    return get_attr_or_default(entity, attr, None)
+
+
+def str_to_bool(string):
+    """Convert xs::boolean to python bool."""
+    if not string:
+        return None
+    if string == 'true':
+        return True
+    if string == 'false':
+        return False
+    raise RuntimeError('Expected "true" or "false", got {}'.format(string))
+
+
+def get_dictionary_from_key_value_pairs(pairs):
+    """Get dictionary from key-value pairs."""
+    if not pairs:
+        return None
+    return {pair.name: pair.value for pair in pairs}
+
+
+def evaluate_predicate(predicate):
+    """Evaluate a predicate to true or false."""
+    # TODO(ivanpauno): Implement me please
+    return True
+
+
+def filtrate_with_predicates(items):
+    """Filter out elements depending on predicate condition."""
+    # TODO(ivanpauno): Implement me please.
+    # ret = []
+    # for item in items:
+    #     if_predicate = get_attr_or_none(item, 'if')
+    #     unless_predicate = get_attr_or_none(item, 'unless')
+    #     if evaluate_predicate(if_predicate) and \
+    #        not evaluate_predicate(unless_predicate):
+    #         ret.append(item)
+    # return ret
+    return items
+
+
+def parse_executable(entity: Entity):
+    """Parse executable tag."""
+    cmd = entity.cmd
+    print(cmd)
+    cwd = get_attr_or_none(entity, 'cwd')
+    name = get_attr_or_none(entity, 'name')
+    shell = str_to_bool(get_attr_or_none(entity, 'shell'))
+    prefix = get_attr_or_none(entity, 'launch-prefix')
+    output = get_attr_or_none(entity, 'output')
+    args = get_attr_or_none(entity, 'args')
+    print(args)
+    args = args.split(' ') if args else []
+    if not type(args) == list:
+        args = [args]
+    print(args)
+    env = get_dictionary_from_key_value_pairs(
+        filtrate_with_predicates(get_attr_or_none(entity, 'env')))
+    if_predicate = get_attr_or_none(entity, 'if')
+    unless_predicate = get_attr_or_none(entity, 'unless')
+
+    cmd_list = [cmd]
+    print(cmd_list)
+    cmd_list.extend(args)
+    print(cmd_list)
+    return launch.actions.ExecuteProcess(
+        cmd=cmd_list,
+        cwd=cwd,
+        env=env,
+        name=name,
+        shell=shell,
+        prefix=prefix,
+        output=output)
+
+    # if evaluate_predicate(if_predicate) and not \
+    #    evaluate_predicate(unless_predicate):
+    #     cmd = [cmd].extend(args)
+    #     return launch.ExecuteProcess(
+    #         cmd=cmd,
+    #         cwd=cwd,
+    #         env=env,
+    #         name=name,
+    #         shell=shell,
+    #         prefix=prefix,
+    #         output=output)
+    # return None
+
+
+def get_remap_rules_from_remap_list(remap_list):
```

Comment:
@ivanpauno nit: can't this one be replaced with just `[remap.from, remap.to for remap in remappings]`?